configure.ac

```diff
@@ -337,25 +363,14 @@ if test "x$PRESET_CFLAGS" = "x"; then
   fi
 fi
 
-dnl set squid required flags
-if test "$squid_cv_compiler" = "gcc"; then
-  case "$squid_host_os" in
-  mingw)
-dnl Guido Serassio (serassio@squid-cache.org) 20070811
-dnl Using the latest MinGW (gcc 3.4.5 + mingw-runtime 3.13) cannot build with
-dnl -Werror -Wmissing-prototypes -Wmissing-declarations
-dnl TODO: check if the problem will be present in any other newer MinGW release.
-    SQUID_CFLAGS="$squid_cv_cc_option_wall -Wpointer-arith -Wwrite-strings -Wcomments -Wshadow"
-    ;;
-  *)
-    SQUID_CFLAGS="$squid_cv_cc_option_wall -Wpointer-arith -Wwrite-strings -Wmissing-prototypes -Wmissing-declarations -Wcomments -Wshadow"
-    ;;
-  esac
-  SQUID_CXXFLAGS="$squid_cv_cc_option_wall -Wpointer-arith -Wwrite-strings -Wcomments -Wshadow -Woverloaded-virtual"
-else
-  SQUID_CFLAGS=
-  SQUID_CXXFLAGS=
-fi
+# be very pedantic with warnings, if the compiler supports it
+SQUID_CC_ADD_CXXFLAG_IF_SUPPORTED([-Wpointer-arith],[squid_cv_cc_w_pointer_arith])
+SQUID_CC_ADD_CXXFLAG_IF_SUPPORTED([-Wwrite-strings],[squid_cv_cc_w_write_strings])
+SQUID_CC_ADD_CXXFLAG_IF_SUPPORTED([-Wcomments],[squid_cv_cc_w_comments])
+SQUID_CC_ADD_CXXFLAG_IF_SUPPORTED([-Wshadow],[squid_cv_cc_w_shadow])
+SQUID_CC_ADD_CXXFLAG_IF_SUPPORTED([-Wmissing-prototypes],[squid_cv_cc_w_missing_prototypes])
```

Comment:
`cc1plus: warning: command line option ‘-Wmissing-prototypes’ is valid for C/ObjC but not for C++ [enabled by default]`

Semaphore CI shows these warnings for this option. Remove it? Since we do not have many C files left, adding SQUID_CC_ADD_CFLAG_IF_SUPPORTED() code may not be worth the effort?

Reply:
Sounds good; removing it